conans/test/unittests/client/build/cmake_test.py

```diff
@@ -1052,7 +1049,7 @@ def test_clean_sh_path(self):
         cmake.configure()
         self.assertIn(self.tempdir, conanfile.path)
 
-        cmake.generator = "MinGW Makefiles"
+        cmake = CMake(conanfile, generator="MinGW Makefiles")
```

Comment:
`.generator` is not documented, it should be added to the constructor (other variables like `generator_platform` will take that value into account).

Reply:
I wouldn't change this, it would be better if it were RO, but the truth is that it is not, and we will most likely break someone if we change this one. 

So I would go for the third one, if generator is assigned, call again to ``get_generator_platform`` unless the platform was set explicitly in the constructor too.